api/v1alpha1/clustertemplate_types.go

```diff
@@ -51,26 +51,19 @@ type InputDataSchema struct {
 	// PolicyTemplateSchema
 }
 
-type ClusterTemplateValidation struct {
-	// Says of the ClusterTemplate is valid or not.
-	ClusterTemplateIsValid bool `json:"clusterTemplateIsValid"`
-	// Holds the error in case the ClusterTemplate is invalid.
-	ClusterTemplateError string `json:"clusterTemplateError,omitempty"`
-}
-
 // ClusterTemplateStatus defines the observed state of ClusterTemplate
 type ClusterTemplateStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Tells if the schema provided in spec.inputDataSchema is a valid JSON.
-	ClusterTemplateValidation ClusterTemplateValidation `json:"clusterTemplateValidation"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 
 // ClusterTemplate is the Schema for the clustertemplates API
+// +kubebuilder:validation:XValidation:message="Spec changes are not allowed for the ClusterTemplate that has passed the validation", rule="!has(oldSelf.status) || oldSelf.status.conditions.exists(c, c.type=='ClusterTemplateValidated' && c.status=='False') || oldSelf.spec == self.spec"
```

Comment:
nit - wording: ...for a ClusterTemplate that has passed validation...

Reply:
Will update in another PR